garage/tf/regressors/base2.py

```diff
@@ -0,0 +1,175 @@
+"""Regressor base classes without Parameterized."""
+import abc
+
+import tensorflow as tf
+
+from garage.misc.tensor_utils import flatten_tensors
+from garage.misc.tensor_utils import unflatten_tensors
+
+
+class Regressor2(abc.ABC):
+    """
+    Regressor base class without Parameterized.
+
+    Args:
+        input_shape (tuple): Input shape.
+        output_dim (int): Output dimension.
+        name (str): Name of the regressor.
+    """
+
+    def __init__(self, input_shape, output_dim, name):
+        self._input_shape = input_shape
+        self._output_dim = output_dim
+        self._name = name
+        self._variable_scope = None
+        self._cached_params = {}
+        self._cached_param_dtypes = {}
+        self._cached_param_shapes = {}
+        self._cached_assign_ops = {}
+        self._cached_assign_placeholders = {}
+
+    def fit(self, xs, ys):
+        """
+        Fit with input data xs and label ys.
+
+        Args:
+            xs: Input data.
+            ys: Label of input data.
+        """
+        raise NotImplementedError
+
+    def predict(self, xs):
+        """
+        Predict ys based on input xs.
+
+        Args:
+            xs: Input data.
+        """
+        raise NotImplementedError
+
+    def get_params_internal(self, **tags):
```

Comment:
Why are we still keeping `get_params_internal`? I thought they are used mainly for `Parameterized`

Reply:
The optimizers require those API from `Parameterized`, like `get_params`, `get_param_values` and `set_param_values`. So I keep them in order to make it compatible with the optimizers.

Reply:
I see. But since these are not used with the optimizers yet, can we just remove them as other primitive? (Or if other primitives have them, I am ok as long as they are consistent)

Reply:
They do. `GaussianMLPBaseline` uses `GaussianMLPRegressor` and optimizers require `GaussianMLPBaseline` to have these API, e.g. in PPO. We already have one test running with `PPO` and `GaussianMLPBaselineWithModel`.

I think other primitives also need them, if they will be used by optimizers.